controllers/config.go

```diff
@@ -0,0 +1,100 @@
+package ocibuilder
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/ocibuilder/ocibuilder/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// watches configuration for controller
+func (ctrl *Controller) watchControllerConfigMap(ctx context.Context) (cache.Controller, error) {
+	ctrl.logger.Info("watching controller config map updates")
+	source := ctrl.newControllerConfigMapWatch()
+	_, controller := cache.NewInformer(
+		source,
+		&corev1.ConfigMap{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if cm, ok := obj.(*corev1.ConfigMap); ok {
+					ctrl.logger.Info("detected configmap update. updating the controller config")
+					err := ctrl.updateConfig(cm)
+					if err != nil {
+						ctrl.logger.Error("update of config failed", "err", err)
+					}
+				}
+			},
+			UpdateFunc: func(old, new interface{}) {
+				if newCm, ok := new.(*corev1.ConfigMap); ok {
+					ctrl.logger.Info("detected configmap update. updating the controller config.")
+					err := ctrl.updateConfig(newCm)
+					if err != nil {
+						ctrl.logger.WithError(err).Error("update of config failed")
+					}
```

Comment:
Same as above, can be reduced. Also, if we receive an error here we should be returning